packages/check/match.js

```diff
@@ -352,6 +366,24 @@ const testSubtree = (value, pattern) => {
       path: '',
     };
   }
+  
+  if (pattern instanceof OnlyOneOf) {
+    const results = [];
+    for (let i = 0; i < pattern.choices.length; ++i) {
+      results.push(testSubtree(value, pattern.choices[i]));
+    }
+
+    if (results.filter(r => r === false).length === 1) {
+
+      // Only one succeeded? Yay, return.
+      return false;
+    }
```

Comment:
It'd be better to count the passed tests in the loop and throw early. In other words, if there are 10 tests and all pass, it should evaluate only the first two.